gstd/src/common/handlers.rs

```diff
@@ -25,50 +25,208 @@
 //! debug and non-debug mode, for programs built in `wasm32` architecture.
 //! For `debug` mode it provides more extensive logging.
 
+#[cfg(target_arch = "wasm32")]
 #[cfg(feature = "oom-handler")]
 #[alloc_error_handler]
 pub fn oom(_: core::alloc::Layout) -> ! {
     crate::ext::oom_panic()
 }
-
+/// We currently support 3 panic handler profiles:
+/// - `panic-handler`: it displays `panicked with '<unknown>'`
+/// - `panic-message`: it displays `panicked with '{message}'`
+/// - `panic-location`: it displays `panicked with '{message}' at '{location}'`
+///
+/// How we get the panic message in different versions of Rust:
+/// - In nightly Rust, we use `#![feature(panic_info_message)]` and the
+///   [`write!`] macro.
+/// - In stable Rust, we need to modify the default panic handler message
+///   format.
+///
+/// Default panic handler message format (according to <https://github.com/rust-lang/rust/pull/112849>):
+/// - Rust  <1.73: `panicked at '{message}', {location}`
+/// - Rust >=1.73: `panicked at {location}:\n{message}`
```

Comment:
Should be removed because we force Rust 1.73

Reply:
I'll remove <1.73 but keep the current format to document the code itself

Reply:
Docs about elder rust still are here